test/adapter/json_api/relationship_test.rb

```diff
@@ -152,18 +152,20 @@ def test_relationship_with_everything
 
         def test_relationship(expected, test_options = {})
           parent_serializer = AuthorSerializer.new(@author)
-
+          association_serializer = @serializer
           serializable_resource_options = {} # adapter.instance_options
 
-          meta = test_options.delete(:meta)
-          options = test_options.delete(:options)
-          links = test_options.delete(:links)
-          association_serializer = @serializer
+          options = test_options.delete(:options) || {}
+          options[:new_options] = {
+            meta: test_options.delete(:meta),
+            links: test_options.delete(:links),
+            reflection: parent_serializer._reflections.to_a.first
```

Comment:
I know this is broken (in the rebase off of master), but just wanted to push the work for the moment